frontend/snapps/httpsnapp/src/index.ts

```diff
@@ -0,0 +1,74 @@
+import fetch from 'node-fetch';
+
+type PublicKey = string;
+
+type Signature = { field: string; scalar: string };
+
+type Signed<Signable> = {
+  publicKey: PublicKey;
+  signature: Signature;
+  payload: Signable;
+};
+
+type Method = 'GET' | 'POST';
+
+type QueryParams = {};
+
+type Response = {
+  ok?: Signed<string>;
+  error?: Error;
+};
+
+/**
+ * Currently, WIP. Sends either a GET or POST request to the HTTPSnapps Oracle API which will
+ * returned a signed response from an official oracle key.
+ *
+ * @param _method
+ * @param _url
+ * @param _params
+ * @param _postData
+ * @returns
+ */
+export const verify = async (
+  method: Method,
+  url: string,
+  params: QueryParams,
+  _postData: string
+): Promise<Response> => {
+  // Convert QueryParams to query string
+  const qs = Object.keys(params)
```

Comment:
We could consider [this method](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams):
```js
const qs = new URLSearchParams(params);
```

Reply:
Good catch switched it out. Thanks!